### ISSUE TITLE
MatchData#[] with an unbounded Range not supported

### ISSUE THREAD

**Issue by postmodern** [oracle/truffleruby#2755]
It does not appear that `MatchData#[]` supports accepting an unbounded range of indices.

## Steps To Reproduce

```ruby
string = "aaaabababab"
regexp = /(ab)(ab)(ab)/
match = string.match(regexp)
p match[1..]
```

### Expected Result

```ruby
["ab", "ab", "ab"]
```

### Actual Result

```
<internal:core> core/type.rb:276:in `convert_type': no implicit conversion of Range into Integer (TypeError)
	from <internal:core> core/type.rb:178:in `rb_to_int_fallback'
	from (irb):4:in `[]'
	from (irb):4:in `<top (required)>'
	from <internal:core> core/kernel.rb:407:in `loop'
	from <internal:core> core/throw_catch.rb:36:in `catch'
	from <internal:core> core/throw_catch.rb:36:in `catch'
	from /opt/rubies/truffleruby-22.2.0/lib/gems/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from <internal:core> core/kernel.rb:376:in `load'
	from <internal:core> core/kernel.rb:376:in `load'
	from /opt/rubies/truffleruby-22.2.0/bin/irb:42:in `<main>'
```

## Version Information

* truffleruby 22.2.0, like ruby 3.0.3, GraalVM CE Native [x86_64-linux]

**Comment by nirvdrum** [comment]
Unfortunately, that error message isn't entirely helpful. We do support ranges, but only when both the start and end values are specified.

```ruby
string = "aaaabababab"
regexp = /(ab)(ab)(ab)/
match = string.match(regexp)
p match[1..3] # => ["ab", "ab", "ab"]
```

There's nothing in the Ruby Spec Suite for this, so it's likely an oversight.

**Comment by andrykonchin** [comment]
Thank you for report. Fixed in b7d7cd0890896672c587bbb9277b00f7bb468c70.
